File: modules/coupon-domain/src/main/java/com/thanh0x/coursedeal/dto/CouponSummaryDTO.java
package com.thanh0x.coursedeal.dto;

import com.thanh0x.coursedeal.model.coupon.CourseLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;

/**
 * Optimized DTO for coupon list view.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CouponSummaryDTO {
    private int courseId;
    private String title;
    private String category;
    private CourseLevel level;
    private String author;
    private float rating;
    private int students;
    private String previewImage;
    private String couponUrl;
    private Instant expiredDate;
    private int usesRemaining;
    private boolean isNew;
}

File: modules/coupon-domain/src/main/java/com/thanh0x/coursedeal/dto/CouponSummaryDTO.kt
package com.thanh0x.coursedeal.dto

import com.thanh0x.coursedeal.model.coupon.CourseLevel
import java.time.Instant

/**
 * Optimized DTO for coupon list view.
 */
data class CouponSummaryDTO(
    val courseId: Int,
    val title: String? = null,
    val category: String? = null,
    val level: CourseLevel? = null,
    val author: String? = null,
    val rating: Float = 0f,
    val students: Int = 0,
    val previewImage: String? = null,
    val couponUrl: String? = null,
    val expiredDate: Instant? = null,
    val usesRemaining: Int = 0,
    @get:JvmName("isNew")
    val isNew: Boolean = false
)
